logstash-core/lib/logstash/plugin.rb

```diff
@@ -122,12 +123,22 @@ def metric
                          LogStash::Instrument::NamespacedNullMetric.new(@metric, :null)
                        end
   end
+
   # return the configured name of this plugin
   # @return [String] The name of the plugin defined by `config_name`
   def config_name
     self.class.config_name
   end
 
+  # commit event to dlq
+  def dlq_commit(event, reason)
```

Comment:
I see the point about codecs and see it useful (depending on the codec). If the codec is complicated and throws an error we may want to DLQ this. That raises the question, if the event is not fully formed, what then?

I'd say we leave this in here for now since a codec has to actually implement using the DLQ.